src/ekf.cpp

```diff
@@ -362,7 +362,26 @@ namespace RobotLocalization
     estimateErrorCovariance_ = (transferFunctionJacobian_ *
                                 estimateErrorCovariance_ *
                                 transferFunctionJacobian_.transpose());
-    estimateErrorCovariance_.noalias() += (processNoiseCovariance_ * delta);
+
+    // Compute processNoiseCovariance scaling if nessesary
+    Eigen::MatrixXd processNoiseCovariance = processNoiseCovariance_;
```

Comment:
Why do we need a copy? I mean, why don't we modify `processNoiseCovariance_` directly?